src/main/java/org/broadinstitute/consent/http/models/FileStorageObject.java

```diff
@@ -137,6 +153,18 @@ public void setBlobId(BlobId blobId) {
         this.blobId = blobId;
     }
 
+    /**
+     * Computes the last time this file was changed, be it created, updated, or deleted.
+     * @return The last time the file has been changed.
+     */
+    public Instant getLatestUpdateDate() {
+        List<Instant> dates = new ArrayList<>();
+        dates.add(this.getCreateDate());
+        dates.add(this.getUpdateDate());
+        dates.add(this.getDeleteDate());
+        return dates.stream().filter(Objects::nonNull).max(Instant::compareTo).orElse(this.getCreateDate());
```

Comment:
To follow up on Elliot's feedback, yes, it's true that there is a non-null so you're guaranteed a create date. Create date is already in your list, so the `.orElse(this.getCreateDate())` condition is moot. You can probably fix both concerns by using a new instant (now) in the or else case.

Reply:
So the weird thing about `Stream.of` and `List.of` is that they both have `@NonNull` constraints... the reason I took the approach I did was because of the fact that `List.of` was throwing me null pointer errors when there was no delete or update date. But, strangely, `Stream.of` seems prefectly happy to take `Stream.of(null, null, null)`, even though it has `@NonNull`. Strange... I like this approach a lot more though.